[refactor] move more functions to InsertObserversHelper

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26696 [refactor] move more functions to InsertObserversHelper**
* #26642 [quant][graphmode] Skip observing bias across function call hierarchy
* #26641 [refactor] Add ObserveHelper and remove some common function parameters
* #26640 [quant][graphmode][fix] remove unneeded code

Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17558701](https://our.internmc.facebook.com/intern/diff/D17558701)

jerryzh168 committed:
[refactor] move more functions to InsertObserversHelper

Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -243,28 +221,56 @@ Node* InsertObserversHelper::insertObserverFor(
-    qconfig = qconfig_dict.at(key);
-  } else {
-    qconfig = parent_qconfig;
+void InsertObserversHelper::findValuesInPattern(
```
The name of the function doesn't reflect its details. It takes many assumption and does a very specific thing - looks for a value in the pattern with a specific name "output" and adds this to a map. We need to call out these assumptions in the comments and better reflect it in the name. Also, I think it might be worth renaming "output" to "intermediate" or something like that.

jerryzh168 replied:
sounds good

ZolotukhinM reviewed (approved):
LGTM!

jerryzh168 committed:
Update on "[refactor] move more functions to InsertObserversHelper"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
